crates/common/consensus/src/electra/beacon_block.rs

```diff
@@ -55,6 +56,29 @@ impl SignedBeaconBlock {
                 .into(),
         })
     }
+
+    pub fn get_blob_sidecars(
+        signed_block: SignedBeaconBlock,
+        blobs: Vec<Blob>,
+        blob_kzg_proofs: Vec<KZGProof>,
+    ) -> anyhow::Result<Vec<BlobSidecar>> {
+        let mut blob_sidecars = Vec::with_capacity(blobs.len());
+
+        for (index, blob) in blobs.into_iter().enumerate() {
+            let blob_and_proof = BlobAndProofV1 {
+                blob,
+                proof: *blob_kzg_proofs
+                    .get(index)
+                    .ok_or_else(|| anyhow!("Kzg_proof not available for blob at index: {index}"))?,
+            };
+
+            let blob_sidecar = signed_block.blob_sidecar(blob_and_proof, index as u64)?;
+
+            blob_sidecars.push(blob_sidecar);
+        }
+
+        Ok(blob_sidecars)
+    }
```

Comment:
```suggestion
    pub fn get_blob_sidecars(
        signed_block: SignedBeaconBlock,
        blobs: Vec<Blob>,
        blob_kzg_proofs: Vec<KZGProof>,
    ) -> anyhow::Result<Vec<BlobSidecar>> {
        blobs
        .into_iter()
        .zip(blob_kzg_proofs.into_iter())
        .enumerate()
        .map(|(index, (blob, proof))| {
            signed_block.blob_sidecar(BlobAndProofV1 { blob, proof }, index as u64)
        })
        .collect::<anyhow::Result<Vec<_>>>()
    }
```

we can simplify this to the above